Automattic/woocommerce-payments
# fix: pricing decimal formatting for tokenized cart

Fixes partially https://github.com/Automattic/woocommerce-payments/issues/8373

#### Changes proposed in this Pull Request









The store's currency number of decimals are not taken into account when providing the data to GooglePay/ApplePay.
We need to ensure we're providing cents to the GooglePay/ApplePay API.

Fixing.

Before:
<img width="1195" alt="Screenshot 2024-06-20 at 5 22 38 PM" src="https://github.com/Automattic/woocommerce-payments/assets/273592/59fae8d9-f8f6-4d4b-844c-0a5c443cffc2">
<img width="1140" alt="Screenshot 2024-06-20 at 5 22 21 PM" src="https://github.com/Automattic/woocommerce-payments/assets/273592/61fefc6e-368a-4c9e-9e13-6c4de4146c82">

After:
<img width="1400" alt="Screenshot 2024-06-20 at 5 23 46 PM" src="https://github.com/Automattic/woocommerce-payments/assets/273592/a0ea5ffc-8239-49ec-b658-a9c68a1b2ef0">
<img width="1312" alt="Screenshot 2024-06-20 at 5 23 13 PM" src="https://github.com/Automattic/woocommerce-payments/assets/273592/e6b79ee1-1431-47d1-b5ed-9d61ac734ed9">

#### Testing instructions





- Enable the "Enable Cart-Token implementation for PRBs" flag in your dev tools (you'll need to update them)
- Ensure you have Google Pay/Apple Pay enabled in the merchant's settings
- Go to WooCommerce > Settings > General > Number of decimals, change the decimals from `2` to `0` (for example - you could also try `1`)
- Go to a product page
- Click on the GooglePay/ApplePay button
- The amounts displayed in the GooglePay/ApplePay dialog should always have two decimals (regardless of the currency)
- Exit the dialog, add the product to the cart, go to classic cart or checkout
- Click on the PRB button
- The amounts displayed in the GooglePay/ApplePay dialog should always have two decimals (regardless of the currency)
  - PLEASE NOTE: this still doesn't work too well for certain currencies that don't have decimals, like JPY
<img width="613" alt="Screenshot 2024-06-20 at 5 39 21 PM" src="https://github.com/Automattic/woocommerce-payments/assets/273592/3bd042b2-51c2-4d7a-aa0e-9c48bffb8f2a">


-------------------


**Post merge**
